addons/web/static/src/js/views/file_upload_progress_bar.js

```diff
@@ -0,0 +1,76 @@
+odoo.define('web.ProgressBar', function (require) {
+'use strict';
+
+const { _t } = require('web.core');
+const Dialog = require('web.Dialog');
+const Widget = require('web.Widget');
+
+const ProgressBar = Widget.extend({
+    template: 'web.ProgressBar',
+
+    events: {
+        'click .o_upload_cross': '_onClickCross',
+    },
+
+    /**
+     * @override
+     * @param {Object} param1
+     * @param {String} param1.title
+     * @param {String} param1.fileUploadId
+     * @param {XMLHttpRequest} param2.xhr
+     */
+    init(parent, { title, fileUploadId, xhr }) {
+        this._super(...arguments);
+        this.title = title;
+        this.fileUploadId = fileUploadId;
+        this.xhr = xhr;
+    },
+
+    /**
+     * @override
+     * @return {Promise}
+     */
+    start() {
+        this.xhr.onabort = () => this.do_notify(_t("Upload canceled"));
```

Comment:
should call `update()`, so that width is correctly computed. This should prevent some flickering when switching folders while uploading some files